R/idr0021 - Segmentation.ipynb

```diff
@@ -0,0 +1,702 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Compute the features (measurements)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "shapes = computeFeatures.shape(threshImg)\n",
+    "moments = computeFeatures.moment(threshImg)\n",
+    "features <- merge(shapes, moments, by=0, all=TRUE)\n",
+    "features"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Save the results back to OMERO"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Create ROIs"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can create an Ellipse for each object taking the x,y coordinates (m.cx and m.cy), the radii (this is the m.majoraxis and calculate the 'minor' axis from 'major' axis and the eccentricity (see [documentation](https://www.rdocumentation.org/packages/EBImage/versions/4.14.2/topics/computeFeatures) of the computeFeatures method)) and the angle theta by which the Ellipse might be rotated.\n",
+    "\n",
+    "In order to be able to re-use this code later, we define a function for it:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#' Create ROIs from a features table\n",
+    "#'\n",
+    "#' @param features The shape and moments generated by EBImage computeFeatures\n",
+    "#' @return A dataframe specifying the x, y, rx, ry, w, h, theta and text parameters of the ROIs\n",
+    "createROIs <- function(features) {\n",
+    "    rois <- data.frame(x = c(0), y = c(0), rx = c(0), ry = c(0), w = c(0), h = c(0), theta = c(0), text = c('remove'), stringsAsFactors = FALSE)\n",
+    "    for (index in 1:length(features[,1])) {\n",
+    "        x <- features[index,8]\n",
+    "        y <- features[index,9]\n",
+    "        r1 <- features[index,10]\n",
+    "        ecc <- features[index,11]\n",
+    "        r2 <- sqrt(r1^2 * (1 - ecc^2))\n",
+    "        theta <- features[index,12]\n",
+    "        rois <- rbind(rois, c(x, y, r1, r2, NA, NA, -theta, as.character(index)))\n",
+    "    }\n",
+    "    rois <- rois[-1,]\n",
+    "    rownames(rois) <- c()\n",
+    "    return(rois)\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "rois <- createROIs(features)\n",
+    "rois"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Save the ROIs back to OMERO"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "addROIs(image, coords = rois)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Exercise:** Open the image in OMERO.web and compare the ROIs with the segmented image."
```

Comment:
I would not compare the ROIs with the segmented image. The segmented image is somewhere else (in the Jupyter notebook). Also, why not to use the OMERO.iviewer itself, the image is there with the ROIs on it, so I can see immediately the quality of my ROIs ?

When I opened my ROIs in iviewer, I noticed they are not associated with any channel. Is that a feature you are using later ?

Reply:
That's what I actually meant, just open the image in OMERO.web (iviewer) and have a look at the ROIs. 

Just realized I've not implemented the 'setChannel' functionality in the R gateway yet, so I can't associate the channel with ROI yet.

Reply:
>That's what I actually meant, just open the image in OMERO.web (iviewer) and have a look at the ROIs.

@dominikl If this is so, then just remove the words ``segmented image``. The image in iviewer will not be segmented.